File: 0014.js
var longestCommonPrefix = function(strs) {
    if (strs.length == 0){
        return ""
    }
    let finalResult = ""
    const shortest_word = strs.reduce(function(a, b) {
        return a.length <= b.length ? a : b;
        })
    for(i = shortest_word.length; i > 0; i--){
        const res = strs.map( string => string.startsWith(shortest_word.slice(0,i)))
        if(res.every(Boolean)){
            finalResult= shortest_word.slice(0,i)
            break
        }
    }
    return finalResult
};



console.log(longestCommonPrefix(["flower","flow","flight"]))
console.log(longestCommonPrefix(["dog","racecar","car"]))
console.log(longestCommonPrefix([]))

File: solutions/014/0014.py
class Solution:
    def longestCommonPrefix(self, strs: list[str]) -> str:

        if len(strs) == 0:
            return ""
        else:
            shortest_word = min(strs, key = len)

            for ind in reversed(range(1,len(shortest_word)+1)):
                if all([y.startswith(shortest_word[:ind]) for y in strs ]):
                    return shortest_word[:ind]
            return ""

x = Solution()
print(x.longestCommonPrefix(strs=["flower","flow","flight"]))
print(x.longestCommonPrefix(strs=[]))
print(x.longestCommonPrefix(strs=["ft"]))

# x = ["flower","flow","flight"]




        